cognite_toolkit/_cdf_tk/utils/fileio/_readers.py

```diff
@@ -0,0 +1,55 @@
+from abc import ABC, abstractmethod
+from collections.abc import Iterator, Mapping
+from io import TextIOWrapper
+from pathlib import Path
+
+from cognite_toolkit._cdf_tk.exceptions import ToolkitValueError
+from cognite_toolkit._cdf_tk.utils._auxiliary import get_concrete_subclasses
+from cognite_toolkit._cdf_tk.utils.collection import humanize_collection
+from cognite_toolkit._cdf_tk.utils.useful_types import JsonVal
+
+from ._base import FileIO
+from ._compression import COMPRESSION_BY_SUFFIX, Compression
+
+
+class FileReader(FileIO, ABC):
+    def __init__(self, input_file: Path) -> None:
+        self.input_file = input_file
+
+    def read_chunks(self) -> Iterator[JsonVal]:
+        compression = Compression.from_filepath(self.input_file)
+        with compression.open("r") as file:
+            yield from self._read_chunks_from_file(file)
+
+    @abstractmethod
+    def _read_chunks_from_file(self, file: TextIOWrapper) -> Iterator[JsonVal]:
+        """Read chunks from the file."""
+        raise NotImplementedError("This method should be implemented in subclasses.")
+
+    @classmethod
+    def from_filepath(cls, filepath: Path) -> "FileReader":
+        suffix = filepath.suffix
+        if suffix in COMPRESSION_BY_SUFFIX:
+            if len(filepath.suffixes) > 1:
+                suffix = filepath.suffixes[-2]
+            else:
+                raise ToolkitValueError(
+                    f"File has a compression suffix, but no file format suffix found. Available formats: {humanize_collection(COMPRESSION_BY_SUFFIX.keys())}."
+                )
+
+        if suffix in FILE_READ_CLS_BY_FORMAT:
+            return FILE_READ_CLS_BY_FORMAT[suffix](input_file=filepath)
+        raise ToolkitValueError(
+            f"Unknown file format: {suffix}. Available formats: {humanize_collection(FILE_READ_CLS_BY_FORMAT.keys())}."
+        )
+
+
+FILE_READ_CLS_BY_FORMAT: Mapping[str, type[FileReader]] = {}
+for subclass in get_concrete_subclasses(FileReader):  # type: ignore[type-abstract]
```

Comment:
Can you add some tests for this code path?

Reply:
This is part of PRs to introduce new functionality. As this is just a base class, it does not make sense to introduce tests for this. For example, when I add the records read/write classes I have this tests that covers this part of the code: https://github.com/cognitedata/toolkit/pull/1802/files#diff-8fca510d461329e4b2707e116ae1146f4dcb77a6536acf852abc92c290c71531R226

Reply:
I understand your reasoning, but I'd prefer that we add dedicated tests for the base class itself.

While the code will be covered by the subclass tests, a base class should not depend on its children for testing. Relying on subclass tests makes our test suite more brittle. For example, if those subclasses are changed or removed in the future, we could silently lose test coverage for this core logic.

Reply:
Ok, added a few tests.